main/ertp/api/purse.md

```diff
@@ -1,78 +1,147 @@
-const purse = issuer.makeEmptyPurse();
-const payment = mint.mintPayment(amountMath.make(123));
-const bucks123 = amountMath.make(123);
+const { issuer: quatloosIssuer, mint: quatloosMint, amountMath: quatloosAmountMath } = 
+      makeIssuerKit('quatloos');
+const quatloosPurse = quatloosIssuer.makeEmptyPurse();
+const payment = quatloosMint.mintPayment(quatloosAmountMath.make(123));
+const quatloos123 = quatloosAmountMath.make(123);
 
-// Deposit a payment for 123 bucks into the purse. Ensure that this is the amount you expect.
-purse.deposit(payment, bucks123);
+// Deposit a payment for 123 Quatloos into the purse. 
+const depositAmountA = quatloosPurse.deposit(payment, quatloos123);
 
-const secondPayment = mint.mintPayment(amountMath.make(100));
+const secondPayment = quatloosMint.mintPayment(quatloosAmountMath.make(100));
 // Throws error
-purse.deposit(secondPayment, fungible123);
+const depositAmountB = quatloosPurse.deposit(secondPayment, quatloos123);
 
 ```
 
 ## purse.makeDepositFacet()
 
 Creates a deposit-only facet on the `purse`. This is an object you can give to other parties
-that lets them deposit to your  `purse` without being able to withdraw, making it a safe
-way to let other people send you payments.
+that lets them deposit `payments` to your  `purse` without being able to withdraw assets from or check
+the balance of the `purse`. This makes it a safe way to let other people send you `payments`.
+
+You can only deposit a `payment` into a deposit facet that's the same `brand` as the original `purse`
+takes.
+ 
 ```js
 const depositOnlyFacet = purse.makeDepositFacet();
-// Give depositOnlyFacet to someone else. They can pass a payment that will be deposited:
+
+// Give depositOnlyFacet to someone else. Anyone with a deposit facet reference can tell it to receive
+// a payment, thus depositing the payment assets in the purse associated with the deposit facet.
 depositOnlyFacet.receive(payment);
 ```
+Once you have created a `depositFacet`, there is one method you can call 
+on it. `depositFacet.receive(payment)`. The `depositFacet` takes the `payment` 
+and adds it to the balance of the facet's associated `purse`. The `payment` 
+must be of the same `brand` as what the `purse` holds.
+
+Note the difference in method names for adding assets between a `purse` and its `depositFacet`.
+To add assets to a `purse` directly, you use `purse.deposit()`. To add assets
+to a `purse` via its `depositFacet`, you use `depositFacet.receive()`.
 
 ## purse.withdraw(amount)
 - `amount` `{Amount}`
 - Returns: `{Payment}`
 
-Withdraw the `amount` from this purse into a new Payment.
+Withdraw the `amount` specified digital assets from this `purse` into a new `payment`.
+
+If the call succeeds, it immediately extracts the value into a new `payment`. 
+The caller won't get the new `payment` until a later turn, since the call is (nearly always) remote.
+But as soon as the message is processed, the value is gone from the `purse`.
```

Comment:
Hmm, where is this coming from? if we are calling it in this way `purse.withdraw`, then we are assuming the purse is local, so the caller gets the result synchronously, in the same turn. If we are thinking the purse is remote, then we need to call it like `E(purse).withdraw` I think it's best to stick to the purse being local, which means we should remove this
```suggestion
```

Reply:
It's coming from a Keybase conversation with Chris about one of his comments. 
Me: "You made a comment in purse.md " I think the fact that deposit() uniquely requires a non-promise (and the implications) should be called out explicitly in the entry for deposit()."   What are the implications, so I can add them to the deposit() section? Thanks."

Reply:
Ok, I'm wondering if we should be adding so much new material when we're just trying to get this out. This new material has nuances, as I mentioned above. In the rest of the doc, we describe the purse as if it is local. Chris' analysis assumes it is remote. We will need to explain why we are suddenly shifting perspective.

Reply:
Why would it make sense to present the API as if most interactions are with a local purse? There are certainly aspects of the interaction for which the distinction isn't relevant but nearly all interactions are with remote purses, so giving developers an awareness of the asynchronous behavior where it's pertinent seems important.

Reply:
It's certainly correct that most interactions will be with a remote purse. I'm just pointing out that this document, with this new  inclusion, will be remarkably inconsistent. We need to explain that this particular analysis is for a remote purse, and our instructions previously (in which we mention calling `purse.deposit` or `purse.withdraw`) only work on local purses. If the purse is remote, we must do `E(purse).deposit` and `E(purse).withdraw`. It's the inconsistency that is the problem, not the analysis of remote purses.